### PR TITLE
Add status bar item for JSON schema selection

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
Add new status bar item in VSCode status bar. This status bar shows current schema used and allow to select/switch schema with UI (quick pick box). It also indicate with error background if yaml-ls uses more than one schema for file.

Demo:

https://user-images.githubusercontent.com/929743/141985832-d207df10-9a26-48c1-99cb-f9edd327180b.mov

Depends on https://github.com/redhat-developer/yaml-language-server/pull/556

### What issues does this PR fix or reference?

Resolve: redhat-developer/vscode-yaml#565

### Is it tested? How?
With test

### CODE REVIEW

File: src/schema-status-bar-item.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      statusBarItem.tooltip = 'Select JSON Schema';
+      statusBarItem.backgroundColor = undefined;
+    } else if (schema.length === 1) {
+      statusBarItem.text = schema[0].name ?? schema[0].uri;
+      statusBarItem.tooltip = 'Select JSON Schema';
+      statusBarItem.backgroundColor = undefined;
+    } else {
+      statusBarItem.text = 'Multiple JSON Schemas...'; //schema.map((val) => val.name ?? val.uri).join(' | ');
+      statusBarItem.tooltip = 'Multiple JSON Schema used to validate this file, click to select one';
+      statusBarItem.backgroundColor = new ThemeColor('statusBarItem.errorBackground');
```
Comment (gorkem) [inline src/schema-status-bar-item.ts:76]: Is this an error situation or a warning one? Should we use `warningBackground` instead?

File: src/schema-status-bar-item.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    if (schema.length === 0) {
+      statusBarItem.text = 'No JSON Schema';
+      statusBarItem.tooltip = 'Select JSON Schema';
+      statusBarItem.backgroundColor = undefined;
+    } else if (schema.length === 1) {
+      statusBarItem.text = schema[0].name ?? schema[0].uri;
+      statusBarItem.tooltip = 'Select JSON Schema';
+      statusBarItem.backgroundColor = undefined;
+    } else {
+      statusBarItem.text = 'Multiple JSON Schemas...'; //schema.map((val) => val.name ?? val.uri).join(' | ');
```
Comment (gorkem) [inline src/schema-status-bar-item.ts:74]: remove unused comment

File: src/schema-status-bar-item.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  for (const key in settings) {
+    if (Object.prototype.hasOwnProperty.call(settings, key)) {
+      const element = settings[key];
+
+      if (Array.isArray(element)) {
+        const filePatterns = element.filter((val) => val !== fileUri);
+        settings[key] = filePatterns;
+      }
+
+      if (typeof element === 'string') {
```
Comment (gorkem) [inline src/schema-status-bar-item.ts:156]: do you really need this  when the next if is strict equality?

**Review by gorkem — CHANGES_REQUESTED** [review]
The build seems to fail as well.

File: test/json-schema-selection.test.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import * as chai from 'chai';
+import { createJSONSchemaStatusBarItem } from '../src/schema-status-bar-item';
+import { CommonLanguageClient } from 'vscode-languageclient';
+import * as vscode from 'vscode';
+import { TestLanguageClient } from './helper';
+import * as jsonStatusBar from '../src/schema-status-bar-item';
+const expect = chai.expect;
+chai.use(sinonChai);
+
+describe('Completion should work in multiple different scenarios', () => {
```
Comment (JPinkney) [inline test/json-schema-selection.test.ts:17]:
```suggestion
describe('Status bar should work in multiple different scenarios', () => {
```

**Review by JPinkney — APPROVED** [review]

**Review by gorkem — CHANGES_REQUESTED** [review]
I suppose this should be tested together with https://github.com/redhat-developer/yaml-language-server/pull/556. However I have not been able to compile the project. When I trie the instructions on [CONTRIBUTING.md](https://github.com/redhat-developer/vscode-yaml/blob/main/CONTRIBUTING.md#developing-the-client-and-server-together) compilation fails. When I try to do `yarn link` compilation fails. At the moment, I can not test this change.

**Comment by evidolob** [comment]
@gorkem Could you share errors which you have?

I try several times and all compilation pass for me.

Other problem with this PR that tests on CI is failing for some reasons, locally they pass. Going to investigate that.

**Comment by evidolob** [comment]
@gorkem Now flow with `yarn link` should work, the problem was with replacement of `process` lib.

**Review by gorkem — APPROVED** [review]

**Pull request merged**